### ISSUE TITLE
Calling sorted() on list[int] leads to C compilation error

### ISSUE THREAD

**Issue by plajjan** [actonlang/acton#1129]
### Acton Version

0.14.2.20221226.12.1.16

### Reproduction code

```
actor main(env):
    l = [7, 4, 9, 1]
    s = sorted(l)
    if s != [1, 4, 7, 9]:
        print(s)
        await async env.exit(1)
```

### Error output

```
ERROR: internal compiler error: compilation of generated C code failed
NOTE: this is likely a bug in actonc, please report this at:
NOTE: https://github.com/actonlang/acton/issues/new?template=ice.yaml
NOTE: acton 0.14.2.20221226.12.1.16 compiled by ghc 8.10 on linux x86_64
NOTE: cc: 0.10.0
cc stdout:

cc stderr:
actonc-71f022d948a7dc14/out/types/sorted.c:88:52: error: use of undeclared identifier '$sorted'
    self->s = (($list (*) ($Ord, $Iterable, $list))$sorted)(sorted$$w$24, sorted$$w$25, self->l);
                                                   ^
actonc-71f022d948a7dc14/out/types/sorted.c:124:69: warning: incompatible pointer types passing 'sorted$$main' (aka 'struct sorted$$main *') to parameter of type '$Cont' (aka 'struct $Cont *') [-Wincompatible-pointer-types]
    return sorted$$main$methods.__init__($tmp, p$1, $CONSTCONT(p$2, $tmp));
                                                                    ^~~~
/home/kll/terastream/acton/dist/rts/rts.h:135:30: note: passing argument to parameter here
$Cont $CONSTCONT($WORD, $Cont);
                             ^
actonc-71f022d948a7dc14/out/types/sorted.c:124:53: warning: incompatible pointer types passing '$Cont' (aka 'struct $Cont *') to parameter of type '$Env' (aka 'struct $Env *') [-Wincompatible-pointer-types]
    return sorted$$main$methods.__init__($tmp, p$1, $CONSTCONT(p$2, $tmp));
                                                    ^~~~~~~~~~~~~~~~~~~~~
2 warnings and 1 error generated.
```

**Comment by plajjan** [comment]
`$sorted` is missing in builtin_functions.h but fixing that, I instead get a segfault, so this still needs to be looked into. @sydow? :)

**Comment by plajjan** [comment]
list_sort is currently commented out in https://github.com/actonlang/acton/blob/main/builtin/list_impl.c#L380

and the sorted() function just returns NULL: https://github.com/actonlang/acton/blob/main/builtin/builtin_functions.c#L225

so yeah, not implemented yet...

**Comment by sydow** [comment]
Thank you…

It is not an accident or a sign of laziness that the implementation of sorting has not yet been done. To implement a fast, memory-efficient, stable, generic sort is a non-trivial engineering task. There seems to be a consensus that the algorithm to choose is timsort, as first implemented for Python. But CPython’s timsort is around 1500 lines of code (including extensive comments). One can find at least two implementations on GitHub, but I don’t know how reliable they are. I will take a look.

**Comment by plajjan** [comment]
Hehe, you're welcome :)

@sydow, we had an embryo of a sort function, written by you I believe, that referenced some heapsort from libbsd. I'm not sure how much work it is to complete something like that, but can't we start by doing something simple(r) and advance to a better solution later?

**Comment by plajjan** [comment]
Hehe, you're welcome :)

@sydow, we had an embryo of a sort function, written by you I believe, that referenced some heapsort from libbsd. I'm not sure how much work it is to complete something like that, but can't we start by doing something simple(r) and advance to a better solution later?

**Comment by plajjan** [comment]
@sydow there is a timsort in libxml2 which we already depend on. Is it the same timsort or has it been specialized in some way for libxml2 (I don't think so, but can't really tell).

Isn't this our sort function?? https://github.com/GNOME/libxml2/blob/master/timsort.h#L531

**Comment by plajjan** [comment]
@sydow yes, MIT allows any changes to the code (or well, what becomes our derivative work). We just have to keep the copyright notice and MIT license terms in our work as well... so hack on! :)

**Comment by sydow** [comment]
Fixed by #1211
